components/switch/switch-visibility.js

```diff
@@ -32,6 +41,35 @@ class VisibilitySwitch extends LocalizeCoreElement(SwitchMixin(LitElement)) {
 		return html`<d2l-icon icon="tier1:visibility-show"></d2l-icon>`;
 	}
 
+	get text() {
+		if (this.on && this._hasConditions) {
+			return this.localize('components.switch.visibleWithPeriod');
+		}
+		else if (this.on) {
+			return this.localize('components.switch.visible');
+		}
+		else {
+			return this.localize('components.switch.hidden');
+		}
+	}
+
+	_createLabel() {
+		const tooltipHelpClasses = {
+			'conditions-show': this.on && this._hasConditions
+		};
+
+		const conditions = html`
+			<d2l-tooltip-help class="${classMap(tooltipHelpClasses)} d2l-switch-text" id="conditions-help" text="${this.localize('components.switch.conditions')}" inherit-font-style>
+				<slot @slotchange="${this._handleConditionsSlotChange}"></slot>
+			</d2l-tooltip-help>
+			`;
+
+		return html`<span id="${this._textId}" class="d2l-switch-text">${this.text}${conditions}</span>`;
```

Comment:
Is there a way to re-use what we return in `super().render()` and insert the conditions inside of that html literal?

Reply:
Thoughts on having that be a method, rather than a getter? I was thinking of doing 

```js
return html`${super.labelContent()}${conditions}`;
```

So that we're not copy-pasting `this.text`, in case in the future we want to change that. 
But to make use of `super()` within `labelContent()`, we would need it to be a method. 

Admittedly, it would probably also be fine we just copy-pasted `${this.text}` and kept it as a getter,l as suggested above.